protocol/contracts/oracle/src/state/supported_pairs.rs

```diff
@@ -234,26 +232,22 @@ fn check_tree(
         {
             Err(ContractError::DuplicatedNodes {})
         } else {
-            Ok(())
+            tree.iter().try_for_each(|swap| {
```

Comment:
what about switching to `Iterator::all` - it sounds closer to the intent here, ah?

Reply:
`Iterator::all` doesn't support `Result`s, only `bool`s. While it might be more suitable for duplication check, I won't be able to use it here. At most I might be able to use `Iterator::find`.

Reply:
no, then let's continue with `try_for_each`